관리자(admin) 페이지 구현

jma1020 opened the pull request:
# Pull Request
- [x] Feature

# Summary
관리자 페이지 중 **유저** 관리 및 **권한** 관리 페이지 구현

# Describe your changes
- 기존 axios api 코드 삭제
- **권한**, **유저** 관리 페이지 react-query 사용
- **권한** 추가 form  react hook form 사용
- **유저** 관리 페이지 유저 List 무한스크롤 구현 (useInfiniteQuery 및 IntersectionObserver 사용)
- Select 컴포넌트 사용


# Issue number and link
#57 

# ETC
에러 및 로딩처리는 회의 후 작성 예정

jma1020 committed:
feat: admin 페이지 api 함수 정의

- admin 페이지에서 사용할 api 함수 정의

jma1020 committed:
feat: admin 페이지

- admin 권한 페이지 구현
- admin 유저리스트 페이지 구현
- admin 게시글리스트 페이지 구현

jma1020 committed:
fix: 간단한 스타일 수정

jma1020 committed:
feat: 관리자 페이지 user 블랙 및 권한변경

- user 블랙 api 연결
- 권한변경 select 컴포넌트 생성

jma1020 committed:
feat: 관리자 페이지 userCard 컴포넌트 타입 수정

jma1020 committed:
feat: admin api roleUser return 명시

jma1020 committed:
fix: 간단한 스타일 수정

jma1020 committed:
fix: selectRoles를 부모로부터 props를 받는 형태로 변경

jma1020 committed:
feat: 기존 menuObj ->  constants 이동

- constants로 이동을 하여 컴포넌트에  의존성이 있었던 함수 파라미터로
  전달

jma1020 committed:
feat: admin api 검색 기능 추가에 따른 api 수정

jma1020 committed:
feat: admin 페이지 user 권한 및 black user UI 구성

jma1020 committed:
feat: admin 페이지 user 검색 기능 추가

jma1020 committed:
feat: admin 페이지 post 임시 데이터 생성

jma1020 committed:
feat: menu toggle BackGround 설정

- BackGround 설정으로 인해 외부 영역 클릭 시 toggle close 효과 추가

jma1020 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/admin-page

jma1020 committed:
feat: user 컴포넌트 react query

jma1020 committed:
docs: admin 파일 구조 및 네이밍형식  변경

jma1020 committed:
fix: type, utils ts config 경로 지정

jma1020 committed:
fix: React query 세팅  오류 수정

jma1020 committed:
refact: role , search-params 타입 분리

jma1020 committed:
feat: Role 컴포넌트 React Query 적용 및 리팩토링

jma1020 committed:
refact: getRoles , getUsers api 형식 변경

- React Query 형식에 맞게 변경

jma1020 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/admin-page

jma1020 committed:
feat: 어드민 페이지 유저 및 유저card 컴포넌트 생성

jma1020 committed:
refact: ManagedUser type 분리

jma1020 committed:
feat: admin User 페이지 List 컴포넌트 추가 구현 및 React Query 적용

jma1020 committed:
refact: getRoles, getUsers React Query 형식에 맞게 변경

jma1020 committed:
fix: 알 수 없는 주석제거

jma1020 committed:
refact: ManagedUserSearch컴포넌트 리팩토링

jma1020 committed:
fix: getUsers api 수정 및 타입 지정

jma1020 committed:
fix: ManagedUserQuery hook 디렉터리 변경

jma1020 committed:
refact: useToggle 사용으로 styled  props 네이밍 변경

jma1020 committed:
refact: type 수정 및 네이밍 컨벤션에 맞게 변수명 수정

jma1020 committed:
refact: 비동기 처리 로직 삭제 및 리팩토링

- react query 사용으로 인해 기존 비동기 처리 로직 삭제
- active toggle state ->  useToggle hook 으로 변경
- searchParams state 생성

jma1020 committed:
refact: ManagedUser List 컴포넌트 리팩토링

- useInfinityQuery를 사용
- IntersectionObserver 코드 추가

jma1020 committed:
refact: ManagedUserSearch 컴포넌트 리팩토링

- styled 파일 분리
- searchParams props에 따른 코드 리팩토링

jma1020 committed:
refact: ManagedUserCard 컴포넌트 리팩토링 및 useBlackUser hook 생성

- type 수정 및 useToggle 사용
- blackUser api useMutate 사용 및 hook 생성

jma1020 committed:
fix: 간단한 코드 수정 및 사용하지 않는 코드 삭제

jma1020 committed:
refact: 사용하지 않는 코드 삭제

jma1020 committed:
refact: useQuery 사용으로 인한 리팩토링

jma1020 committed:
refact: User 권한 변경 api -> useMutation 으로 변경

jma1020 committed:
fix: 사용하지 않는 import 및 코드 삭제

jma1020 committed:
fix: import 컨벤션으로 인한 수정

jma1020 committed:
feat: BlackUser도 무한스크롤 형식으로 변경

- useInfinity query 사용
- IntersectionObserver 사용

jma1020 committed:
refact: styled 파일 분리 및 간단한 style 수정

jma1020 committed:
refact: 사용하지 않는 import 삭제

jma1020 committed:
refact: ManagedUserCard  Modal 코드 분리

jma1020 committed:
fix: SearchUserParams -> SearchParams 로 네이밍 변경

jma1020 committed:
fix: admin api 타입 인터페이스 네이밍 변경

jma1020 committed:
refact: styled 컴포넌트 네이밍 변경

jma1020 committed:
refact: searchParams state page 삭제

- useInfinity query에서 pageParam 사용으로 인한 page 프로퍼티 삭제

jma1020 committed:
refact: searchParams state page 삭제

jma1020 committed:
feat: 게시글 관리 페이지 api response type 생성

seongjin2427 reviewed (approved):
고생하셨습니다!

jma1020 committed:
refact: getPost api 형식 변경

- react query 형식에 맞도록 변경

jma1020 committed:
refact: ManagedPost 컴포넌트 리팩토링

- ManagedPostList 에서 data 핸들링으로 인해 기존 api 요청 코드 삭제
- 임시 데이터 삭제
- useToggle 사용
- searchParams 상태 생성

jma1020 committed:
feat: ManagedPostList 컴포넌트 생성 및 React query 사용

- PostList와 ReportPostList 를 가져오는 useQuery hook 생성

jma1020 committed:
feat: ManagedPostCard 컴포넌트 구현

- 각 게시물 삭제 및 작성자 블랙 mutation 적용
- 게시물 삭제 mutation hook 생성
- useModal, useToggle 사용

jma1020 committed:
feat: ManagedPostSearch 컴포넌트 생성

- 게시물 검색 컴포넌트 생성
- Select 컴포넌트 사용

sangrimlee commented on api/admin.ts:
```diff
@@ -0,0 +1,196 @@
+  roleId: number;
+  roleName: string;
+  description: string;
+}
```
타입 분리을 분리하면 좋을 것 같아요.
interface에 대한 네이밍 컨벤션도 같이 고민해보면 좋을 것 같아요.

jma1020 replied:
이번 회의 때 정한 컨벤션 적용해서 수정하겠습니다!!😃

sangrimlee commented on components/admin/ManagedPost/ManagedPost.tsx:
```diff
@@ -0,0 +1,76 @@
+  };
+  useEffect(() => {
+    getAllPost();
+  }, []);
```
react-query를 적용하는 건 추후에 적용하실 예정인가요?

jma1020 committed:
refact:ManagedUserList컴포넌트  네이밍 컨벤션에 따른 변경

jma1020 committed:
refact: ManagedUserCard 컴포넌트 네이밍 컨벤션에 따른 수정

jma1020 committed:
refact: ManagedUserRoleChange 컴포넌트 네이밍 컨벤션에 따른 수정

jma1020 committed:
refact: ManagedUserSearch 컴포넌트 네이밍 컨벤션에 따른 변경

jma1020 committed:
refact: 컨벤션에 따른 간단한 수정

jma1020 committed:
refact: 컨벤션에 따른 간단한 수정

jma1020 committed:
refact: 컨벤션에 따른 간단한 수정

jma1020 committed:
refact: 컨벤션에 따른 간단한 수정

jma1020 committed:
refact: 컨벤션에 따른 수정 및 type, 파일  네이밍 변경

jma1020 committed:
fix: 로딩 GIF 스피너 삭제 ->추후 SVG로 생성

jma1020 committed:
refact: Role 컴포넌트 mutation hook 분리

jma1020 committed:
refact: Post Menu 추가 및 추가에 따른 네이밍 변경

jma1020 committed:
refact: admin api 타입분리 및 react query 형식으로 통일화

jma1020 committed:
feat: ManagedUser 컴포넌트 context 적용

- 기존 searchParams state를 context로 변경

jma1020 committed:
feat: ManagedPost 컴포넌트 context 적용

- 기존 searchParams state를 context 적용

jma1020 committed:
fix: styled 컴포넌트 tag 네이밍 변경

jma1020 committed:
fix: dropdown 네이밍 변경에 따른 constants 네이밍 수정

jma1020 committed:
fix: styled 컴포넌트 tag 네이밍 변경

sangrimlee commented on components/admin/ManagedPost/contexts/PostSearchParamsContex.tsx:
```diff
@@ -0,0 +1,39 @@
+import React, {
```
파일명이 잘못되었어요!

jma1020 replied:
감사합니다 ㅎㅎ!

sangrimlee commented on types/search-params.ts:
```diff
@@ -0,0 +1,6 @@
+export interface SearchParams {
+  searchType?: string;
+  keyword?: string;
+  page?: number;
+  size?: number;
+}
```
SearchParams 이름을 명시적으로 바꾸어주면 좋을 것 같아요!

jma1020 replied:
수정했습니다!!😀

sangrimlee commented on components/admin/ManagedUser/contexts/useSearchParamsContext.ts:
```diff
@@ -0,0 +1,18 @@
+    throw Error;
+  }
+
+  if (context === undefined) {
+    throw Error;
+  }
```
context가 `undefined`일 경우가 있을까요?

있다면, `if (!context) {}` 로 처리하면 좋을 것 같아요!

jma1020 replied:
앗 error 찾다가 추가했던거네요,, ㅎㅎ  지우고서 올리겠습니다!

jma1020 committed:
fix: 잘못 작성한 파일명 수정 및 Context 수정

- 분리했던 useContext hook 다시 결합
- 컴포넌트 이름 throw 형식으로 변경

jma1020 committed:
fix: 잘못 작성된 파일명 수정 및 context 수정

- 분리했던 useContext hook 다시 결합
- context 네이밍 변경
- 컴포넌트 이름 throw

jma1020 committed:
refact: SearchParams 네이밍 변경 및 ManagedPost styled 컨벤션에 따른
수정

- SearchParams 잘못된 혼용을 막기위해 더 명시적인 네이밍으로 수정
- Styled Container 컨벤션에 따른 컴포넌트명을 추가해 수정

jma1020 committed:
refact: styled 컨벤션에 따른 네이밍 변경

jma1020 committed:
refact: 타입 분리

jma1020 committed:
fix: context error 메세지 수정

jma1020 committed:
refact: admin api type 명시 및 불필요한 console 삭제

jma1020 committed:
Merge branch 'develop' into feature/admin-page
